### ISSUE TITLE
GFortran 10 Argument Mismatch

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS#206]
Just as in ADIOS2: https://github.com/ornladios/ADIOS2/issues/2228

@michaelkuhn reported an issue in Spack, which indicated that ADIOS needs the Fortran compiler flag `-fallow-argument-mismatch` flag in order to build with GCC/GFortran 10.

Note:
```
Using this option is strongly discouraged. It is possible to provide standard-conforming code which allows different types of arguments by using an explicit interface and TYPE(*). 
```
https://gcc.gnu.org/onlinedocs/gfortran/Fortran-Dialect-Options.html

**Describe the bug**
Please see https://github.com/spack/spack/pull/16569

**To Reproduce**
s. above

**Desktop (please complete the following information):**
 - OS/Platform: Ubuntu
 - Build: GCC 10
 - ADIOS version: 1.13.1

**Additional context**
See: https://github.com/spack/spack/pull/16569

**Comment by NAThompson** [comment]
Yeah, I just hit this as well; I couldn't figure out a robust fix. My workaround was just

```
FCFLAGS="-g -O2 -fallow-argument-mismatch" ./configure
```

which I suppose is fairly obvious to anyone who works with `configure` scripts with any regularity.

**Comment by ax3l** [comment]
Yes, We were wondering if there is an actual change that we can implement in the code to address the raised diagnostics instead of muting it.
